contrib/core-contract-tests/tests/pox-4/pox-4.prop.test.ts

```diff
@@ -0,0 +1,1605 @@
+              [Cl.uint(reward_cycle)],
+              account
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.UInt));
+            assert(isClarityType(first_burn_block_height, ClarityType.UInt));
+            assert(isClarityType(reward_cycle_length, ClarityType.UInt));
+            const expected =
+              Number(first_burn_block_height.value) +
+              Number(reward_cycle_length.value) * reward_cycle;
+            expect(actual).toBeUint(expected);
+          }
+        )
+      );
+    });
+
+    it("should return correct burn-height-to-reward-cycle", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          fc.nat(),
+          (account, burn_height) => {
+            // Arrange
+            const { result: pox_4_info } = simnet.callReadOnlyFn(
+              "pox-4",
+              "get-pox-info",
+              [],
+              account
+            );
+            assert(isClarityType(pox_4_info, ClarityType.ResponseOk));
+            assert(isClarityType(pox_4_info.value, ClarityType.Tuple));
+            const first_burn_block_height =
+              pox_4_info.value.data["first-burnchain-block-height"];
+            const reward_cycle_length =
+              pox_4_info.value.data["reward-cycle-length"];
+            // Act
+            const { result: actual } = simnet.callReadOnlyFn(
+              "pox-4",
+              "burn-height-to-reward-cycle",
+              [Cl.uint(burn_height)],
+              account
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.UInt));
+            assert(isClarityType(first_burn_block_height, ClarityType.UInt));
+            assert(isClarityType(reward_cycle_length, ClarityType.UInt));
+            const expected = Math.floor(
+              (burn_height - Number(first_burn_block_height.value)) /
+                Number(reward_cycle_length.value)
+            );
+            expect(actual).toBeUint(expected);
+          }
+        )
+      );
+    });
+
+    it("should return u0 current-pox-reward-cycle", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          (caller) => {
+            // Arrange
+            let expected = 0;
+            // Act
+            const { result: actual } = simnet.callReadOnlyFn(
+              "pox-4",
+              "current-pox-reward-cycle",
+              [],
+              caller
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.UInt));
+            expect(actual).toBeUint(expected);
+          }
+        )
+      );
+    });
+
+    it("should return none get-stacker-info", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          fc.constantFrom(...simnet.getAccounts().values()),
+          (stacker, caller) => {
```

Comment:
What happens is `stacker` and `caller` are the same account?

Reply:
Nothing happens. The reason there are two generated values is to cover both the case when the `caller` and the `stacker` argument are the same and the one they are identical. Should I keep only the `caller` one?